### PR TITLE
Fixed dob backup deploy

### LINKED ISSUE #5: Deployment pipeline for DOB backup service
Discussed in meet with @shikharish

### PR DESCRIPTION
**Pull request by proffapt**
fixes #5

### CODE REVIEW

**Comment by rajivharlalka** [comment]
@proffapt have fixed the workflow and the pgpass config that was needed, but in retrospect I feel this is a very bad idea to have the secret exposed in the image as the image is public and hence the secret. Since this is a build-time env variable, I see two ways keeping this a secret.
1. use docker secrets ( I have little knowledge and 0 will-power to understand how to configure it with this github action)
2. build the image locally on the server (either manually or using an updated github action)

**Comment by proffapt** [comment]
Yes, I too thought about it if that thing is going to be public with those build args. I would first like to understand why is there a need of such build args. I believe things are done in complicated way which could have rather been simplified. Let's hop on a call with @shikharish and understand the architecture and see whether the core concept can be updated instead of us trying to do ninja techniques on docker secrets.

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -52,6 +52,29 @@ jobs:
...
+                  key: ${{ runner.os }}-buildx-dob-backup-${{ github.sha }}
+                  restore-keys: |
+                      ${{ runner.os }}-buildx-dob-backup-
+
+            - name: Build & Push DOB Backup
+              uses: docker/build-push-action@v5
+              with:
+                  context: ./backup
+                  push: true
+                  build-args: PGPASS=${{secrets.PGPASS}}
```
Comment (proffapt) [inline .github/workflows/deploy.yaml:68]: Need to remove this, no build args shall be required.

File: backup/Dockerfile [hunk 2]
```diff
@@ -18,10 +18,10 @@ RUN curl -fsSLO "$SUPERCRONIC_URL" \
 
 WORKDIR /root
 
-ARG DBPASSFILE
+ARG PGPASS
 
-COPY ./backup ./backup
-COPY ${DBPASSFILE} ./.pgpass
+COPY . ./backup
+RUN echo "$PGPASS" >> ./.pgpass
```
Comment (proffapt) [inline backup/Dockerfile:24]: No pgpass shall be created during docker container build

**Comment by proffapt** [comment]
Migrated to #7

**Pull request closed without merging**